### ISSUE TITLE
States should have properties

### ISSUE THREAD

**Issue by StoneCypher** [StoneCypher/fsl#124]
When long-form defining a state, allow it to define named properties on the states as values. Then, the logic can look up a given property on the current state without bothering with tables or whatever.

Reverse polymorphism, in a fashion.  Voluntary aspects, ... sort of.

```fsl
machine_name: "chess reduced";

new_game -> p1_turn <=> p2_turn;
[p1_turn p2_turn] -> [p1_win p2_win draw] -> [exit new_game];

[new_game p1_turn p2_turn].{ properties: { game_underway: true } };
[p1_win p2_win draw].{ properties: { game_underway: false } };
```

**Comment by StoneCypher** [comment]
I'm more and more for this

This is in essence a direct implementation of Moore machine logic (that is, the state you're in defines the outputs.)  

So, these are Moore outputs.  Therefore they're a primary tool and should be used.

Transition properties #445 are essentially the Mealy outputs

**Comment by StoneCypher** [comment]
closed in favor of #761
